### PR TITLE
fix: Search bar acting out

### LINKED ISSUE #3307: Search bar acting out
This was once fixed and it's broken again, search results don't update if you clean the search bar and switch between collection / explorer. 

Steps to reproduce: 

1. go to gallery
2. search for random term, such as 'test'
3. click X in the search bar
4. click on collection
5. go back to gallery

As you will see, search term stays 'test' even after page reloads. Ideally, every page reload checks the input in search bar and updates the results accordingly. Right now, only way to change the search results is to do another search 

image

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #3307 
- [ ] Requires deployment <rubick/magick_issue>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image

### CODE REVIEW

**Comment by prachi00** [comment]
Screen Shot 2022-07-02 at 11 45 19 PM
Screen Shot 2022-07-02 at 11 45 28 PM
on searching anything, it gives this error in this api @Jarsen136

**Comment by Jarsen136** [comment]
> Screen Shot 2022-07-02 at 11 45 19 PM
> 
> Screen Shot 2022-07-02 at 11 45 28 PM
> 
> on searching anything, it gives this error in this api @Jarsen136

There is already an open issue with this error. https://github.com/kodadot/nft-gallery/issues/3321
It would not effect the result in this PR.

File: components/rmrk/Collection/List/CollectionList.vue [hunk 1]
```diff
@@ -263,7 +262,7 @@ export default class CollectionList extends mixins(
 
   @Watch('$route.query.search')
   protected onSearchChange(val: string, oldVal: string) {
-    if (shouldUpdate(val, oldVal)) {
+    if (val !== oldVal) {
```
[the remark this answers is not in the document]
Comment (prachi00) [reply components/rmrk/Collection/List/CollectionList.vue:265]: @Jarsen136  ⬆️

[the remark this answers is not in the document]
Comment (prachi00) [reply components/rmrk/Collection/List/CollectionList.vue:265]: @Jarsen136 ⬆️

[the remark this answers is not in the document]
Comment (Jarsen136) [reply components/rmrk/Collection/List/CollectionList.vue:265]:
```
function shouldUpdate(val: T, oldVal: T): boolean {
  return val && val !== oldVal
}
```

No, it's not. If the newValue is equal to the empty string `''`, `shouldUpdate` will return false.

**Review by prachi00 — APPROVED** [review]

**Comment by prachi00** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 51.83 USD/KSM ~ 1.929 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0x8ec04285459339e6f89d59fce9f556cad5a76bc7f9d57df838aee7c9134437d3](https://kusama.subscan.io/extrinsic/0x8ec04285459339e6f89d59fce9f556cad5a76bc7f9d57df838aee7c9134437d3)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**